### PR TITLE
feat: improve MongoDb connection string handling

### PR DESCRIPTION
**Pull request by grtlr**
## Notes to Reviewer

As a reviewer, please pay particular attention to the following areas when reviewing this PR and tick the above boxes after you have completed the steps.

**Config Changes**
* [ ] Ensure proper order in which CLI and config arguments are applied.

### CODE REVIEW

File: src/db/mongodb/mod.rs [hunk 1]
```diff
@@ -35,12 +35,14 @@ impl MongoDb {
...
-                .build();
-            client_options.credential = Some(credential);
+        if client_options.credential.is_none() {
+            if let (Some(username), Some(password)) = (&config.username, &config.password) {
+                let credential = Credential::builder()
+                    .username(username.clone())
+                    .password(password.clone())
+                    .build();
+                client_options.credential = Some(credential);
+            }
```
Comment (Alex6323) [inline src/db/mongodb/mod.rs:45]: Should we error or at least log something if both didn't provide any credentials?

Comment (Alex6323) [reply src/db/mongodb/mod.rs:45]: hmm, probably not necessary as the error will be detailed enough I guess.

**Review by Alex6323 — APPROVED** [review]
lgtm

File: src/db/mongodb/mod.rs (hunk 1 above)
Comment (grtlr) [reply src/db/mongodb/mod.rs:45]: Yeah, I'd think so.

**Pull request merged**